images/cassandra/tests/main.tf

```diff
@@ -0,0 +1,42 @@
+terraform {
+  required_providers {
+    oci     = { source = "chainguard-dev/oci" }
+    kubectl = { source = "gavinbunney/kubectl" }
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = "~> 2.11.0"
+    }
+  }
+}
+
+provider "kubernetes" {
+  config_path    = "~/.kube/config"
+  config_context = "kind-kind"
+}
+
+variable "digest" {
+  description = "The image digest to run tests over."
+}
+
+data "oci_string" "ref" { input = var.digest }
+
+data "oci_exec_test" "test" {
+  for_each = toset(fileset(path.module, "*.sh"))
+
+  digest = var.digest
+  script = "${path.module}/${each.value}"
+}
+
+resource "kubernetes_manifest" "prerequisites" {
+  for_each = fileset(path.module, "manifests/*.yaml")
+  manifest = yamldecode(file("${path.module}/${each.value}"))
+}
+
+resource "kubectl_manifest" "stateful_set" {
```

Comment:
Can we use the `hashicorp/kubernetes` (`kubernetes_stateful_set`) here instead?

The `kubernetes` provider, with all it's faults, does a much better job at integrating into terraforms resource readiness checks than `kubectl` does. Since we're not dealing with any CRDs here, I think that'd give us the best signal of whether this is actually wai.

You can use something like the command below to convert the yaml into HCL:

```
echo 'yamldecode(file("cassandra-statefulset.yaml"))' | terraform console
```

and then you can inline the data references instead of using `templatefile`

Reply:
I think so earlier I was using `hashicorp/kubernetes` for `stateful_set` but there's an open issue with that, we can't use `kubernetes_manifest` as `depends_on` doesnt work I guess with it, here's the [issue link ](https://github.com/hashicorp/terraform-provider-kubernetes/issues/1380#issuecomment-967022975) and what was suggested. I think it was discussed with @jonjohnsonjr yesterday but if you still suggest, I can try again

Reply:
Oh wow, that was TIL, thanks so much, I am still a noob with TF, apologies for the confusion 😅